src/main/java/org/entando/kubernetes/validator/ValidationFunctions.java

```diff
@@ -124,4 +128,58 @@ public static String validateEntityCodeOrThrow(String entityCode) throws Entando
 
         return entityCode;
     }
+
+    /**
+     * Check the URL for correctness. Database connection string are NOT handled by this method
+     * The strategy here is to convert the URL to a URI object and take advantage of the Java library
+     * to detect any error.
+     *
+     * @param candidate the URL to test
+     * @param validateProtocol when false, the check against the protocol is skipped
+     * @param validatePort when false, the port presence is ignored
+     * @param validatePath when false the presence of the path is ignored
+     * @return true if the URL is correct
+     */
+    public static boolean validateURL(String candidate, boolean validateProtocol, boolean validatePort,
+            boolean validatePath) {
+
+        try {
+            candidate = validateProtocol ? candidate : appendProtocolIfMissing(candidate, null);
+
+            final URL url = new URL(candidate);
+            final String port = String.valueOf(url.getPort());
+
+            return ((StringUtils.isNotBlank(url.toURI().toString())
+                    && StringUtils.isNotBlank(url.getProtocol()))
+                    && StringUtils.isNotBlank(url.getHost())
+                    && (!validatePort
+                    || (StringUtils.isNotBlank(port) && StringUtils.isNumeric(port)))
+                    && (!validatePath || StringUtils.isNotBlank(url.getPath())));
+        } catch (Throwable t) {
+            // no op
```

Comment:
pls put a log here, in general you should log more

Reply:
Ok!

Reply:
Done